### PR TITLE
Require at least one version in ThirdPartyResource objects

### LINKED ISSUE #36991: TPR version should not be optional but necessary.
FYI. Currently the `Versions` field of TPR is marked as `optional` which is wrong, actually if you try to create a TPR with empty versions, it will cause api server panic. Versions should be necessary field of TPR, we should validate this somehow. @deads2k

### PR DESCRIPTION
**Pull request by liggitt**
Fixes #36991

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36996)

### CODE REVIEW

**Comment by liggitt** [comment]
on the fence about whether this needs to go in 1.5 or not

**Comment by AdoHe** [comment]
@saad-ali any possibility to get this merged in 1.5?

File: pkg/master/thirdparty/thirdparty.go [hunk 1]
```diff
@@ -246,6 +246,9 @@ func (m *ThirdPartyResourceServer) InstallThirdPartyResource(rsrc *extensions.Th
 	if err != nil {
 		return err
 	}
+	if len(rsrc.Versions) == 0 {
```
Comment (AdoHe) [inline pkg/master/thirdparty/thirdparty.go]: any reason to add check here?

Comment (liggitt) [inline pkg/master/thirdparty/thirdparty.go]: to avoid the panic and allow recovery if a system already had a bad one loaded

Comment (AdoHe) [inline pkg/master/thirdparty/thirdparty.go]: That makes sense, could you please leave a comment here? I would like to get this merged before this #36977

Comment (liggitt) [inline pkg/master/thirdparty/thirdparty.go]: I think the logic and returned error are pretty self-explanatory

**Comment by saad-ali** [comment]
> @saad-ali any possibility to get this merged in 1.5?

The issue it is fixing, was it introduced in 1.5? How bad is the issue it is fixing, can we ship 1.5 with it? How risky is the fix to the rest of the 1.5 release?

**Comment by liggitt** [comment]
> The issue it is fixing, was it introduced in 1.5?

no

> How bad is the issue it is fixing, can we ship 1.5 with it?

it crashes the API server and you have to manually remove data from etcd before you can restart

> How risky is the fix to the rest of the 1.5 release?

extremely low risk

**Comment by saad-ali** [comment]
> extremely low risk

Approved for post-code freeze merge for 1.5

**Pull request merged**